cmd/crane/cmd/pull.go

```diff
@@ -33,16 +36,38 @@ func NewCmdPull(options *[]crane.Option) *cobra.Command {
 		Args:  cobra.MinimumNArgs(2),
 		RunE: func(_ *cobra.Command, args []string) error {
 			imageMap := map[string]v1.Image{}
+			indexMap := map[string]v1.ImageIndex{}
 			srcList, path := args[:len(args)-1], args[len(args)-1]
 			for _, src := range srcList {
-				img, err := crane.Pull(src, *options...)
+				o := crane.GetOptions(*options...)
+				ref, err := name.ParseReference(src, o.Name...)
 				if err != nil {
-					return fmt.Errorf("pulling %s: %w", src, err)
+					return fmt.Errorf("parsing reference %q: %w", src, err)
+				}
+
+				rmt, err := remote.Get(ref, o.Remote...)
+				if err != nil {
+					return err
+				}
+
+				// If we're writing an index to a layout and --platform hasn't been set,
+				// pull the entire index, not just a child image.
+				if format == "oci" && rmt.MediaType.IsIndex() && o.Platform == nil {
+					idx, err := rmt.ImageIndex()
```

Comment:
I'd just add an e2e test like this: https://github.com/google/go-containerregistry/blob/2874338840a65b73444563e6bd3540dd2f9271b0/.github/workflows/e2e.yaml#L63-L65

... but use `gcr.io/distroless/static`.  I'd include a variant with `--platform` as well.